h2o-core/src/main/java/water/H2ONode.java

```diff
@@ -438,7 +439,13 @@ void sendBuffer(){
   
     // Open channel on first write attempt
     private ByteChannel openChan() throws IOException {
-      return H2ONode.openChan(TCPReceiverThread.TCP_SMALL,_socketFactory, _key.getAddress(), _key.getPort());
+      try {
+        return H2ONode.openChan(TCPReceiverThread.TCP_SMALL, _socketFactory, _key.getAddress(), _key.getPort());
+      }catch (ConnectException e){
+        // can't establish connection to this node, report this as closed connection ( i.e the target h2o node is down)
+        _connection_closed = true;
+        throw e;
```

Comment:
Yeah but this is in general communication (openChannel) isn't it? Won't you mark the channel closed after the first exception?

I would think that if you can not open connection to the client you should retry until you hit the timeout. It can be only a temporary failure which should not kill the cluster.